Internal PR for LTI management feature

cmltaWt0 opened the pull request:
Issue: https://github.com/cjlee112/socraticqs2/issues/131

cmltaWt0 committed:
feat(lti): add LtiConsumer model to replace system wide LTI setting

Changing logic need to make carefully due to existing LTI users

cmltaWt0 committed:
feat(lti): add data migration to create first LtiConsumer from Django settings

cmltaWt0 committed:
feat(lti): change LtiConsumer search flow
Changes:
- Courselets user (django_user) creation flow: w/o email djangi_user w/ random
- Decide to remove data migration so now we need to create first LtiConsumer

cmltaWt0 committed:
feat(lti): add `expiration_date` DateField to LtiConsumer for LTI requests w/ `short_term` flag

Chages:
- add acceptance tests for LtiConsumer searching flow
- add `file` handler to Django logger settings - use error level for LTI exceptions
- add migrations file
- update admin.py corresponding to new field

cmltaWt0 committed:
fix(lti): fix TravisCI - add logs folder w/ README file

cmltaWt0 committed:
ref(lti): add additional consumer_key check, add additional logging

cmltaWt0 committed:
doc(lti): add Short-term LTI credentials description to documentation `lti.rst`

derdmitry commented on mysite/lti/utils.py:
```diff
@@ -16,3 +21,37 @@ def wrapped(request, *args, **kwargs):
+    username = ''
+    for _index in range(30):
+        username = username + random.SystemRandom().choice(allowable_chars)
+    return username
```
Why not use some like `uuid.uuid4().hex`?

cmltaWt0 replied:
> Why not use some like uuid.uuid4().hex?

Django username has requirements - up to 30 characters.
We can get only 30 chars from 32 characters uuid.uuid4().hex may be. But I am not sure.

cmltaWt0 replied:
Agreed with you - will change to uuid.uuid4().hex[:30]

cmltaWt0 committed:
ref(lti): remove unnecessary dependency `django-oauth2-provider`

Changes:
- add `key_secret_generator` helper for substitute previously used `provider.utils.short_token`

cmltaWt0 committed:
ref(lti): change random username generation logic,
          update tests for this change,
          substitute last four migrations with only one
